driver/normalizer/normalizer.go

```diff
@@ -23,14 +16,178 @@ var Normalize = Transformers([][]Transformer{
 
 // Preprocessors is a block of AST preprocessing rules rules.
 var Preprocessors = []Mapping{
-	// ObjectToNode defines how to normalize common fields of native AST
-	// (like node type, token, positional information).
-	//
-	// https://godoc.org/gopkg.in/bblfsh/sdk.v2/uast#ObjectToNode
+	// replace "IsEmpty" with "Length: 0" in spans
+	Map(
+		Part("_", Obj{
+			uast.KeyType: String("TextSpan"),
+			"IsEmpty":    AnyNode(nil),
+		}),
+		Part("_", Obj{
+			uast.KeyType: String("TextSpan"),
+			"Length":     Int(0),
+		}),
+	),
+
+	// restore "Start" if it was removed as a default value
+	Map(
+		Check(
+			Not(Has{
+				"Start": AnyNode(nil),
+			}),
+			Obj{
+				uast.KeyType: String("TextSpan"),
+				"Length":     Var("length"),
+				"End":        Var("end"),
+			},
+		),
+		Obj{
+			uast.KeyType: String("TextSpan"),
+			"Start":      Int(0),
+			"Length":     Var("length"),
+			"End":        Var("end"),
+		},
+	),
+
+	// remove SpanStart
+	Map(
+		// TODO(dennwc): add it as a custom position field?
+		Part("_", Obj{
+			"SpanStart": AnyNode(nil),
+		}),
+		Part("_", Obj{}),
+	),
+
+	Map(
+		Part("_", Obj{
+			"FullSpan": Obj{
+				uast.KeyType: String("TextSpan"),
+				"Length":     AnyNode(nil),
+				"Start":      Var("start"),
+				"End":        Var("end"),
+			},
+			// TODO(dennwc): add it as a custom position field?
+			"Span": AnyNode(nil),
+		}),
+		Part("_", Obj{
+			// remap to temporary keys and let ObjectToNode to pick them up
+			"spanStart": Var("start"),
+			"spanEnd":   Var("end"),
+		}),
+	),
+
 	ObjectToNode{
-		InternalTypeKey: "...", // native AST type key name
+		OffsetKey:    "spanStart",
+		EndOffsetKey: "spanEnd",
 	}.Mapping(),
 }
 
 // Normalizers is the main block of normalization rules to convert native AST to semantic UAST.
-var Normalizers = []Mapping{}
+var Normalizers = []Mapping{
+	MapSemantic("IdentifierNameSyntax", uast.Identifier{}, MapObj(
+		Obj{
+			"Identifier": Obj{
+				uast.KeyType: String("SyntaxToken"),
+				// TODO(dennwc): assert that it's the same as in parent
+				uast.KeyPos: AnyNode(nil),
+
+				"LeadingTrivia":  Arr(),
+				"TrailingTrivia": Arr(),
+				"RawKind":        Int(8508),
+				"Text":           Var("name"),
+				"Value":          Var("name"),
+				"ValueText":      Var("name"),
+			},
+			"RawKind": Int(8616),
```

Comment:
May be that is something obvious, but what are the `RawKind` constants for?

Reply:
This is probably an enum for a parser's rule id. I will check if it's possible to change the JSON encoder to get the textual representation for it.

Another question you may ask is why it's mentioned here at all. The reason is the following: the DSL is very strict to fields that appear in AST, but was not covered by the transformation rule. This was done to avoid cases when we silently drop unknown fields from AST nodes. So each rule should list every single field in the AST node either as a variable or as a constant. Unless the transform is `Part`ial, as in most preprocessor transforms.